src/vs/workbench/services/preferences/common/preferencesValidation.ts

```diff
@@ -269,3 +285,51 @@ function getArrayOfStringValidator(prop: IConfigurationPropertySchema): ((value:
 
 	return null;
 }
+
+function getObjectValidator(prop: IConfigurationPropertySchema): ((value: any) => (string | null)) | null {
+	if (prop.type === 'object' && !prop.additionalProperties) {
+		const { properties, patternProperties } = prop;
+		return value => {
+			if (!value) {
+				return null;
+			}
+
+			const errors: string[] = [];
+
+			Object.keys(value).forEach((key: string) => {
+				const data = value[key];
+				if (properties && key in properties) {
+					const errorMessage = getErrorsForSchema(properties[key], data);
+					if (errorMessage) {
+						errors.push(nls.localize('validations.errorsForKey', 'For key "{0}": {1}\n', key, errorMessage));
+					}
+				} else if (patternProperties) {
+					for (const pattern in patternProperties) {
+						if (RegExp(pattern).test(key)) {
+							const errorMessage = getErrorsForSchema(patternProperties[pattern], data);
+							if (errorMessage) {
+								errors.push(nls.localize('validations.errorsForKey', 'For key "{0}": {1}\n', key, errorMessage));
+							}
+							return;
```

Comment:
Seems like it only shows an error for a single key, would  make sense to show all of them here too

Reply:
`getErrorsForSchema` returns all the errors for that key, joined with spaces. I notice that only the string array validator adds in newlines after the error messages, though.

Reply:
All errors for one key, I mean showing errors for all keys. It does that for the case in your screenshot but not for this case